contracts/PartyBid.sol

```diff
@@ -407,6 +406,28 @@ contract PartyBid is ReentrancyGuardUpgradeable, ERC721HolderUpgradeable {
 
     // ============ Internal: Finalize ============
 
+    /**
+    * @notice Query the NFT contract to get the token owner
+    * @dev nftContract must implement the ERC-721 token standard exactly:
+    * function ownerOf(uint256 _tokenId) external view returns (address);
+    * See https://eips.ethereum.org/EIPS/eip-721
+    * @dev Returns address(0) if NFT token or NFT contract
+    * no longer exists (token burned or contract self-destructed)
+    * @return _owner the owner of the NFT
+    */
+    function _getOwner() internal returns (address _owner) {
+        (bool success, bytes memory returnData) =
+            nftContract.call(
+                abi.encodeWithSignature(
+                    "ownerOf(uint256)",
+                    tokenId
+                )
+        );
+        if (success && returnData.length > 0) {
```

Comment:
abi.decode might throw an exception if the `returnData` is the incorrect length (specifically, if it's too short) HOWEVER since we used the same `_getOwner` check in the `initialize` function, that should have surfaced before the party could be created. therefore, we just check that `returnData.length > 0` (e.g. check that the nft contract was not self destructed between `initialize` and `finalize`)

trying to think through other ways to make it even more impossible for this function to throw or revert. you can't use try/catch for non-external calls.